phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/ViewEngines.java

```diff
@@ -159,8 +155,14 @@ private void showEngineDetails(EngineDTO engineDTO) {
         showRightView(currentEnginePanel);
     }
 
-    public void setOnSelectCategory(Consumer<EngineCategoryDTO> onSelectCategory) {
-        this.onSelectCategory = onSelectCategory;
+    private void processFilterText(String filterText) {
```

Comment:
Wouldn't it be possible to pass only the method which shall be called when the search shall start such that the handling of the `pause` and `filterText.toLowerCase()` is inside `SearchBox`?